src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs

```diff
@@ -1902,7 +1902,7 @@ public enum GCS : uint
 
             if (bufferLength > 0)
             {
-                var buffer = bufferLength <= 64 ? stackalloc byte[bufferLength] : new byte[bufferLength];
+                var buffer = bufferLength <= 64 ? stackalloc byte[bufferLength + 8] : new byte[bufferLength];
```

Comment:
Please change the implementation to use `Encoding.Unicode.GetString` so we don't have to append extra length of the buffer. Turns out current implementation isn't optimal.

Reply:
You are right, I changed the implementation in a new commit.